packages/client-core/src/hooks/useEngineCanvas.ts

```diff
@@ -24,17 +24,29 @@ Infinite Reality Engine. All Rights Reserved.
 */
 
 import { getComponent } from '@ir-engine/ecs'
+import { isSupportedBrowser } from '@ir-engine/editor/src/functions/browserCheck'
 import { getState, useHookstate, useMutableState } from '@ir-engine/hyperflux'
 import { EngineState } from '@ir-engine/spatial/src/EngineState'
 import { destroySpatialViewer, initializeSpatialViewer } from '@ir-engine/spatial/src/initializeEngine'
 import { RendererComponent } from '@ir-engine/spatial/src/renderer/WebGLRendererSystem'
 import { useEffect, useLayoutEffect } from 'react'
+import { NotificationService } from '../common/services/NotificationService'
 
 export const useEngineCanvas = (ref: React.RefObject<HTMLElement>) => {
   const lastRef = useHookstate(() => ref.current)
-
   const engineState = useMutableState(EngineState)
 
+  const supportedBrowser = useHookstate(isSupportedBrowser)
```

Comment:
let's move this to something other than the useEngineCanvas, it shouldn't have so many responsibilities

we should make this a hook and use it in the root of the location and studio routes